zerver/views/user_settings.py

```diff
@@ -109,6 +111,29 @@ def json_left_side_userlist(request, user_profile, left_side_userlist=REQ(valida
 
     return json_success(result)
 
+@has_request_variables
+def json_languages(request, user_profile):
+    # type: (HttpRequest, UserProfile) -> HttpResponse
+
+    languages = [
+        {'text': 'English', 'value': 'en'},
+        {'text': '简体中文', 'value': 'zh_CN'},
+    ]
+    return json_success({"languages": languages})
```

Comment:
We should send this down in the do_events_register / page_params data set, rather than having it be a separate AJAX query.  There's a performance benefit to avoiding RTTs for initial data fetching.